source/ShipInfoPanel.cpp

```diff
@@ -468,22 +546,65 @@ void ShipInfoPanel::DrawWeapons(const Rectangle &bounds)
 	Color topColor;
 	bool hasTop = false;
 	auto layout = Layout(static_cast<int>(LABEL_WIDTH), Truncate::BACK);
+
+	weaponsRight.clear();
+	weaponsLeft.clear();
+	indicesRight.clear();
+	indicesLeft.clear();
+	int index = 0;
+
+	// Sort weapons by category (turret or gun).
+	std::vector<const Hardpoint *> gunHardpoints;
+	std::vector<const Hardpoint *> turretHardpoints;
 	for(const Hardpoint &hardpoint : ship.Weapons())
 	{
+		if(hardpoint.IsTurret())
+			turretHardpoints.emplace_back(&hardpoint);
+		else
+			gunHardpoints.emplace_back(&hardpoint);
+	}
+
+	// Sort wepaons into left and right.
+	auto SortIntoIndices = [&] (std::vector<const Hardpoint *> &toSort)
+	{
+		for(const Hardpoint *hardpoint : toSort)
+		{
+			if(hardpoint->GetPoint().X() >= 0.)
+			{
+				indicesRight.emplace_back(index);
+				weaponsRight.emplace_back(hardpoint);
+			}
+			else
+			{
+				indicesLeft.emplace_back(index);
+				weaponsLeft.emplace_back(hardpoint);
+			}
+			index++;
+		}
+	};
+	SortIntoIndices(gunHardpoints);
+	SortIntoIndices(turretHardpoints);
```

Comment:
Wasn't there code already here that sorts the hardpoints into left and right?
Didn't you also sort them into left and right in the other method?
This doesn't seem like a very good way of doing this.
Not only do you seem to be replicating code for doing, if not the same, very similar things across multiple methods, you're running it every frame. This seems especially egregious when the containers you are populating are class fields. It seems counterproductive to store these in the class and then still unnecessarily regenerate them every frame.
The names are also not very informative.

Reply:
Ok, now it is only run alongside UpdateInfo in the same loop as the other stuff, but whats wrong with the names?